command/services/export/export.go

```diff
@@ -0,0 +1,197 @@
+			return 1
+		}
+
+		serviceExists := false
+
+		for i, service := range cfg.Services {
+			if service.Name == c.serviceName {
+				serviceExists = true
+
+				// Add a consumer for each peer where one doesn't already exist
+				for _, peerName := range peerNames {
+					peerExists := false
+					for _, consumer := range service.Consumers {
+						if consumer.Peer == peerName {
+							peerExists = true
+							break
+						}
+					}
+					if !peerExists {
+						cfg.Services[i].Consumers = append(cfg.Services[i].Consumers, api.ServiceConsumer{Peer: peerName})
+					}
+				}
+
+				// Add a consumer for each partition where one doesn't already exist
+				for _, partitionName := range partitionNames {
+					partitionExists := false
+
+					for _, consumer := range service.Consumers {
+						if consumer.Partition == partitionName {
+							partitionExists = true
+							break
+						}
+					}
+					if !partitionExists {
+						cfg.Services[i].Consumers = append(cfg.Services[i].Consumers, api.ServiceConsumer{Partition: partitionName})
+					}
+				}
+			}
+		}
+
+		if !serviceExists {
+			cfg.Services = append(cfg.Services, api.ExportedService{
+				Name:      c.serviceName,
+				Consumers: buildConsumers(peerNames, partitionNames),
+			})
+		}
+
+		ok, _, err := client.ConfigEntries().CAS(cfg, cfg.GetModifyIndex(), nil)
+		if err != nil {
+			c.UI.Error(fmt.Sprintf("Error writing modifed service and peer to config entry: %s", err))
+			return 1
+		}
+
+		if !ok {
+			c.UI.Error(fmt.Sprintf("Modifed service and peer config entry was changed during update. Please try again"))
+			return 1
+		}
+	}
+
+	if len(c.peerNames) > 0 && len(c.partitionNames) > 0 {
+		c.UI.Info(fmt.Sprintf("Successfully exported service %q to peers %q and to partitions %q", c.serviceName, c.peerNames, c.partitionNames))
+	} else if len(c.peerNames) > 0 {
+		c.UI.Info(fmt.Sprintf("Successfully exported service %q to peers %q", c.serviceName, c.peerNames))
+	} else if len(c.partitionNames) > 0 {
+		c.UI.Info(fmt.Sprintf("Successfully exported service %q to partitions %q", c.serviceName, c.partitionNames))
+	}
+	return 0
+}
+
+func buildConsumers(peerNames []string, partitionNames []string) []api.ServiceConsumer {
+	var consumers []api.ServiceConsumer
+	for _, peer := range peerNames {
+		consumers = append(consumers, api.ServiceConsumer{
+			Peer: peer,
+		})
+	}
+	for _, partition := range partitionNames {
+		consumers = append(consumers, api.ServiceConsumer{
+			Partition: partition,
+		})
+	}
+	return consumers
+}
+
+//========
+
+func (c *cmd) Synopsis() string {
+	return synopsis
+}
+
+func (c *cmd) Help() string {
+	return flags.Usage(c.help, nil)
+}
+
+const (
+	synopsis = "Export a service"
+	help     = `
+Usage: consul services export [options] -name <service name> -consumer-peers <other cluster name>
+
+  Export a service. The peers provided will be used locally by
+  this cluster to refer to the other cluster where the services will be exported. 
```

Comment:
nit: I think there some naunce between `cluster name`, which confused with datacenter and `peer name`, which is a local idea.
```suggestion
Usage: consul services export [options] -name <service name> -consumer-peers <peer names>

  Export a service. The peer names provided will be used locally by
  this cluster to refer to the peer where the services will be exported. 
```